service/controller/resource/prometheusremotewrite/delete.go

```diff
@@ -31,27 +33,58 @@ func (r *Resource) EnsureDeleted(ctx context.Context, obj interface{}) error {
 		}
 
 		for _, current := range prometheusList.Items {
-
-			// remove remotewrite config from Prometheus once RemoteWrite CR is deleted
-			if desired, ok := removePrometheusRemoteWrite(*remoteWrite, *current); ok {
-				if !ok {
-					r.logger.Debugf(ctx, fmt.Sprintf("no update required for Prometheus CR %#q in namespace %#q", desired.Name, desired.Namespace))
-					continue
-				}
-				r.logger.Debugf(ctx, fmt.Sprintf("updating Prometheus CR %#q in namespace %#q", desired.Name, desired.Namespace))
-				updateMeta(current, desired)
-				_, err = r.prometheusClient.MonitoringV1().
-					Prometheuses(current.GetNamespace()).
-					Update(ctx, desired, metav1.UpdateOptions{})
-				if err != nil {
-					return microerror.Mask(err)
-				}
+			err = r.unsetRemoteWrite(ctx, remoteWrite, current)
+			if err != nil {
+				return microerror.Mask(err)
 			}
+		}
 
+		err = r.ensureCleanUp(ctx, remoteWrite, prometheusList.Items)
+		if err != nil {
+			return microerror.Mask(err)
 		}
 
 	}
 	r.logger.Debugf(ctx, "deleted prometheus remoteWrite config")
 
 	return nil
 }
+
+func (r *Resource) unsetRemoteWrite(ctx context.Context, remoteWrite *pmov1alpha1.RemoteWrite, prometheus *promv1.Prometheus) error {
+	err := r.ensureStatusDeleted(ctx, remoteWrite, prometheus)
```

Comment:
Same as before, I would suggest putting this call after updating the Prometheus CR